src/messaging/models.py

```diff
@@ -5,19 +5,26 @@
 # Create your models here.
 
 class Thread(models.Model):
-	profile_1 = models.ForeignKey(Profile, null=True, on_delete=models.SET_NULL, related_name='profile_1')
-	profile_2 = models.ForeignKey(Profile, null=True, on_delete=models.SET_NULL, related_name='profile_2')
+    profile_1 = models.ForeignKey(Profile, null=True, on_delete=models.SET_NULL, related_name='profile_1')
+    profile_2 = models.ForeignKey(Profile, null=True, on_delete=models.SET_NULL, related_name='profile_2')
+
+    @staticmethod
+    def getProfileQuery(prof1, prof2=None):
+        query = Q(profile_1=prof1) | Q(profile_2=prof1)
+        if prof2 is not None:
+            query &= Q(profile_1=prof2) | Q(profile_2=prof2)
+        return query
+
+    def get_other_user(self, user):
+        if self.profile_1 == user or self.profile_1.id == user:
```

Comment:
Why check for both?

Reply:
most functions allow taking both an id or an object.  

i'm now realizing if the user is neither profile_1 or profile_2, this would be weird.